Alpha/Base.py

```
@@ -239,49 +240,60 @@ def update(self, algorithm: QCAlgorithm, data: Slice) -> List[Insight]:
+        
+        if not self.check_market_schedule():
+            return []
+        
+        chain = self.dataHandler.getOptionContracts(data)
+        if chain is None:
+            return []
+        
+        insights = self.CreateInsights(chain, data=data)
+        
         self.context.executionTimer.stop('Alpha.Base -> Update')
         return Insight.Group(insights)
```

Ruff (0.8.0)

`Insight` may be undefined, or defined from star imports

(F405)